runtime/src/main/java/org/corfudb/runtime/view/replication/ChainReplicationProtocol.java

```diff
@@ -128,6 +125,49 @@ public ILogData peek(RuntimeLayout runtimeLayout, long globalAddress) {
         return waitOrHoleFill(runtimeLayout, readResult, waitForWrite);
     }
 
+    public void commitAll(RuntimeLayout runtimeLayout, Collection<Long> addresses) {
+        // Group addresses by log unit tail server endpoint.
+        Map<String, List<Long>> serverAddressMap =
+                groupAddressByLogUnitTail(runtimeLayout.getLayout(), addresses);
+
+        // Send read requests to log unit servers in parallel.
```

Comment:
nit -> not read requests... inspect address requests.

Reply:
Done.